packages/drivers/driver-base/src/documentDeltaConnection.ts

```diff
@@ -310,17 +310,25 @@ export class DocumentDeltaConnection
     }
 
     /**
-     * Disconnect from the websocket, and permanently disable this DocumentDeltaConnection.
+     * Disconnect from the websocket and close the websocket too.
+     */
+    protected disposeSocket(error: IAnyDriverError) {
+        this.disposeCore(error);
+    }
+
+    /**
+     * Disconnect from the websocket, and permanently disable this DocumentDeltaConnection and close the socket.
+     * However the OdspDocumentDeltaConnection differ in dispose as in there we don't close the socket. There is no
+     * multiplexing here, so we need to close the socket here.
      */
     public dispose() {
-        this.disposeCore(
-            false, // socketProtocolError
-            createGenericNetworkError(
-                // pre-0.58 error message: clientClosingConnection
-                "Client closing delta connection", { canRetry: true }, { driverVersion }));
+        this.disposeCore(createGenericNetworkError(
```

Comment:
Take a look if we can remove disposeCore in favor of having only dispose() with default value for an error.
I have a feeling that we can simply have something like that:

```ts
protected dispose(err?: IAnyDriverError) {
    if (err === undefined) {
       err = createGenericNetworkError(...);
    }
    ...
}
```

Reply:
No actually not. Here we implement IDisposable where the dispose function takes Error type argument while here the error type is IAnyDriverError. So we can just take the argument as Error type and then case it to IAnyDriverError but that is forceful conversion as it may not overlap and the caller would not know that it should be of IAnyDriverError type.

So mayne a bigger change is required if it needs to be done. Not doing it in this PR as it is not that big of a change,.